File: examples/visibility_timeout/reader.rb
require 'rubygems'
require 'karait'
require 'yaml'

puts "Starting ruby reader."

messages_read = 0.0
start_time = Time.now().to_f

queue = Karait::Queue.new
while true
  messages = queue.read :routing_key => 'for_reader', :messages_read => 15, :visibility_timeout => 0.1
  messages.each do |message|
    messages_read += 1.0
        
    message.messages_read = messages_read
    message.messages_read_per_second = messages_read / (Time.now().to_f - start_time)
    
    puts "Message Read: \n#{message.to_hash.to_yaml}"
  end
  queue.delete_messages messages
end

File: examples/visibility_timeout/reader.js
var puts = require('sys').puts,
    Queue = require('karait').Queue;

puts("Starting javascript reader.")

messagesRead = 0.0
startTime = (new Date()).getTime() / 1000.0;

new Queue(function(err, queue) {
    if (err) {
        throw err;
    }
    
    (function readMessages() {
        queue.read({routingKey: 'for_reader', visibilityTimeout: 3.0}, function(err, messages) {
            for (var i = 0, message; (message = messages[i]) != null; i++) {
                messagesRead += 1;
                message.messages_read = messagesRead;
                message.messages_read_per_second = messagesRead / ( ( (new Date()).getTime() / 1000.0 ) - startTime );
                
                puts(JSON.stringify(message.toObject()));
            }
            queue.deleteMessages(messages, function() {
                readMessages();
            });
        });
    })();
});